osmoutils/slice_helper.go

```diff
@@ -117,3 +117,38 @@ func GetRandomSubset[T any](slice []T) []T {
 	n := rand.Intn(len(slice))
 	return slice[:n]
 }
+
+// Difference returns the slice of elements that are elements of a but not elements of b.
+func Difference[T comparable](a, b []T) []T {
+	mb := make(map[T]struct{}, len(a))
+	for _, x := range b {
+		mb[x] = struct{}{}
+	}
+	diff := make([]T, 0)
+	for _, x := range a {
+		if _, found := mb[x]; !found {
+			diff = append(diff, x)
+		}
+	}
```

Comment:
what about using slices package?

```golang
diff := []T{}
for _, x := a {
   if !slices.Contains(x, b) {
     diff = append(diff, x)
  }
}
```

Reply:
This will have a polinomial time complexity instead of a linear